### PR TITLE
🐛Token Viewer Fix

### LINKED ISSUE #1464: 🐛TokenViewer Returns Only One Token For A Specific Correlation
## Description

The TokenViewer returns only one token for a specific correlation. 

## Reproduction

- start a process where a service task returns e.g. 'true' with a custom correlation ID 
- edit the process that the service task returns 'false' and start it again with the same custom correlation id
- open the TokenViewer on the LET and select the service task
- see that it shows 'true' instead of 'false'

## Possible solution

Return ALL tokens of an element for the specific process instance ID, not for the correlation ID.

### PR DESCRIPTION
**Pull request by alexanderkasten**
**Changes:**

1. Updates Interfaces
2. Use the Correct Method to Fetch Tokens 
3. Fix Showing all Tokens of a Process Instance ID

**Issues:**

Closes #1464 

PR: #1450 

## How can others test the changes?

look at the linked issue.
you can use this example process for testing:

[test_tokenviewer.bpmn.zip](https://github.com/process-engine/bpmn-studio/files/3182571/test_tokenviewer.bpmn.zip)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: src/modules/inspect/token-viewer/token-viewer.ts [hunk 1]
```diff
@@ -33,9 +33,9 @@ export class TokenViewer {
     this._inspectCorrelationService = inspectCorrelationService;
   }
 
-  public correlationChanged(): void {
-    const correlationWasInitiallyOpened: boolean = this.flowNode === undefined;
-    if (correlationWasInitiallyOpened) {
+  public processInstanceIdChanged(): void {
+    const processInstanceWasInitiallyOpened: boolean = this.flowNode === undefined;
```
Comment (ElRaptorus) [inline src/modules/inspect/token-viewer/token-viewer.ts]: Maybe I'm missing something here, but from the name of this business rule, shouldn't this be something like

```suggestion
    const processInstanceWasInitiallyOpened: boolean = this.flowNode !== undefined;
```

?

Maybe find a better name?

File: src/modules/inspect/token-viewer/token-viewer.ts [hunk 2]
```diff
@@ -85,63 +85,73 @@ export class TokenViewer {
...
+      return tokenEntries;
     }
 
-    return tokenHistoryEntries.map((historyEntry: DataModels.TokenHistory.TokenHistoryEntry, index: number) => {
-      // tslint:disable-next-line no-magic-numbers
-      const payloadAsString: string = JSON.stringify(historyEntry.payload, null, 2);
+    Object.entries(tokenHistoryGroup).forEach(([flowNodeId, tokenHistoryEntries]: [string, Array<DataModels.TokenHistory.TokenHistoryEntry>]) => {
+
+      tokenHistoryEntries.forEach((historyEntry: DataModels.TokenHistory.TokenHistoryEntry, index: number) => {
+        // tslint:disable-next-line no-magic-numbers
```
Comment (ElRaptorus) [inline src/modules/inspect/token-viewer/token-viewer.ts]: Rejoice, for this is a rule that'll get kicked out when switching to ESLint. 😉

File: src/modules/inspect/token-viewer/token-viewer.ts [hunk 3]
```diff
@@ -85,63 +85,73 @@ export class TokenViewer {
...
 
-  private _getRawTokenEntriesForFlowNode(tokenHistoryEntries: Array<DataModels.TokenHistory.TokenHistoryEntry>): Array<IRawTokenEntry> {
-    const elementHasNoToken: boolean = tokenHistoryEntries === undefined;
+  private _getRawTokenEntriesForFlowNode(tokenHistoryGroup: DataModels.TokenHistory.TokenHistoryGroup): Array<IRawTokenEntry> {
+    const tokenEntries: Array<IRawTokenEntry> = [];
+
+    const elementHasNoToken: boolean = tokenHistoryGroup === undefined;
     if (elementHasNoToken) {
-      return [];
+      return tokenEntries;
```
Comment (SteffenKn) [inline src/modules/inspect/token-viewer/token-viewer.ts]: I would prefer to return an empty array here.
If someone later edits this function, they can't simply produce a bug by adding stuff to the array.
Also it's a bit easier to read.

File: src/modules/inspect/token-viewer/token-viewer.ts [hunk 4]
```diff
@@ -2,40 +2,48 @@ import {bindable, inject} from 'aurelia-framework';
...
-    this._inspectCorrelationService = inspectCorrelationService;
+  constructor(tokenViewerService: ITokenViewerService) {
+    this._tokenViewerService = tokenViewerService;
   }
 
-  public correlationChanged(): void {
-    const correlationWasInitiallyOpened: boolean = this.flowNode === undefined;
-    if (correlationWasInitiallyOpened) {
+  public processInstanceIdOrCorrelationChanged(): void {
+    const noFlowNode: boolean = this.flowNode === undefined;
```
Comment (SteffenKn) [inline src/modules/inspect/token-viewer/token-viewer.ts]: You can improve this variable name

```suggestion
    const noFlowNodeSelected: boolean = this.flowNode === undefined;
```

File: src/modules/inspect/token-viewer/token-viewer.ts [hunk 5]
```diff
@@ -2,40 +2,48 @@ import {bindable, inject} from 'aurelia-framework';
...
+    if (noFlowNode) {
```
Comment (SteffenKn) [inline src/modules/inspect/token-viewer/token-viewer.ts]:
```suggestion
    if (noFlowNodeSelected) {
```

**Comment by ElRaptorus** [comment]
This is what the TokenView looks like on my machine, using the most recent version of this branch:

Bildschirmfoto 2019-05-27 um 11 33 32


There is a correlation, but no matter which FlowNode I select, the TokenViewer never shows any tokens.

File: src/modules/inspect/token-viewer/token-viewer.ts [hunk 6]
```diff
@@ -2,40 +2,48 @@ import {bindable, inject} from 'aurelia-framework';
...
   IPayloadEntryValue,
   IRawTokenEntry,
   ITokenEntry,
+  ITokenViewerService,
 } from './contracts/index';
 
-@inject('InspectCorrelationService')
+// tslint:disable: no-magic-numbers
+
+const versionRegex: RegExp = /(\d*)\.(\d*).(\d*)/;
```
Comment (rrrene) [inline src/modules/inspect/token-viewer/token-viewer.ts]: Warum sind das hier `*` und nicht `+`? 

Der obige Regex ergibt auch auf den String `Urmel...2000` einen match ...

**Pull request merged**